packages/calypso-e2e/src/lib/pages/login-page.ts

```diff
@@ -43,75 +54,113 @@ export class LoginPage {
+	 *
+	 * @param param0 LoginCredentials object.
+	 * @param {string} param0.username User name.
+	 * @param {string} param0.password Password.
 	 */
-	async login( { username, password }: { username: string; password: string } ): Promise< void > {
-		await this.pageSettled();
-
-		await this.switchAccount();
-
-		// Begin the process of logging in.
+	private async baseflow( { username, password }: LoginCredentials ): Promise< void > {
 		await this.page.fill( selectors.username, username );
 		await this.page.keyboard.press( 'Enter' );
 		await this.page.fill( selectors.password, password );
-		await this.page.click( selectors.loginButton );
-		// Make sure after logging in that everything stablizes, so we can continue with the next action!
+		// Wait for navigation to resolve in a regular login.
+		await Promise.all( [
+			this.page.waitForNavigation(),
+			this.page.click( selectors.loginButton ),
+		] );
+		// After navigation resolves, wait for `load` state to fire.
 		await this.page.waitForLoadState( 'load' );
 	}
 
+	/* Log in methods */
+
 	/**
-	 * Navigates to the Login page.
+	 * Log in to WordPress.com from the `/log-in` endpoint.
+	 *
+	 * This is the 'normal' or 'standard' way of performing log ins.
+	 *
+	 * @param {LoginCredentials | TestAccount} credentials Credentials of the user. Specify either an username/password pair or name of a test account in the configuration.
 	 */
-	async visit(): Promise< void > {
-		await this.page.goto( getCalypsoURL( 'log-in' ), { waitUntil: 'networkidle' } );
+	async login( credentials: LoginCredentials | TestAccount ): Promise< void > {
+		await this.visit();
+		await this.switchAccount();
+
+		let username;
+		let password;
+
+		if ( 'account' in credentials ) {
+			// Test Account specified. Look for the corresponding username/password
+			// combination from the configuration file.
+			[ username, password ] = getAccountCredential( credentials.account );
+		} else {
+			// Regular username/password pair specified - destructure it.
+			( { username, password } = credentials );
+		}
+
+		await this.baseflow( { username, password } );
 	}
 
+	/* Signup */
+
 	/**
-	 * Clicks on a link to navigate to the account signup page.
+	 * Visit the Login page and click on the signup link for a new account.
 	 */
-	async clickSignup(): Promise< void > {
+	async signup(): Promise< void > {
 		await this.visit();
-		await this.pageSettled();
-
 		await this.switchAccount();
+
 		await Promise.all( [
 			this.page.waitForNavigation(),
 			this.page.click( selectors.createAccountLnk ),
 		] );
 	}
 
+	/* Magic Links */
+
 	/**
 	 * Given an email address or username, request a magic link for the user.
 	 *
+	 * This method requires the user to be on the main Login page.
+	 *
 	 * @param {string} user Username or email address of the user requesting a link.
 	 */
 	async requestMagicLink( user: string ): Promise< void > {
+		await this.visit();
+		await this.switchAccount();
+
```

Comment:
Meant to be an all-encapsulating method that performs required steps to get this done.